http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -630,37 +630,44 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
       // we can nack the message on errors. Monitoring all creations will be asynchronous,
       // and (3) and (4) will always be asynchronous.
 
-      // Create the cluster synchronously
-      createClusterResultOpt <- msg.cluster
-        .traverse { c =>
-          gkeInterp.createCluster(CreateClusterParams(c.clusterId, msg.project, List(c.defaultNodepoolId), false))
-        }
-        .adaptError {
-          case e =>
-            PubsubKubernetesError(
-              AppError(e.getMessage, ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Cluster, None),
-              Some(msg.appId),
-              false,
-              msg.cluster.map(_.defaultNodepoolId),
-              msg.cluster.map(_.clusterId)
-            )
-        }
-
-      // Create the nodepool synchronously if we didn't need to create the cluster
-      createNodepoolResultOpt <- if (createClusterResultOpt.isDefined) F.pure(none[CreateNodepoolResult])
-      else
-        msg.nodepoolId
-          .traverse(nodepoolId => gkeInterp.createNodepool(CreateNodepoolParams(nodepoolId, msg.project)))
-          .adaptError {
-            case e =>
-              PubsubKubernetesError(
-                AppError(e.getMessage, ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.Nodepool, None),
-                Some(msg.appId),
-                false,
-                msg.nodepoolId,
-                None
+      // Create the cluster or nodepool synchronously if necessary
+      monitorClusterOrNodepool <- msg.clusterNodepoolAction match {
+        case Some(ClusterNodepoolAction.CreateClusterAndNodepool(clusterId, defaultNodepoolId, nodepoolId)) =>
+          for {
+            createClusterResult <- gkeInterp
+              .createCluster(
+                CreateClusterParams(clusterId, msg.project, List(defaultNodepoolId, nodepoolId), false)
```

Comment:
Note we're now passing 2 nodepools to the cluster creation request, which is explicitly specified in the message

Reply:
just for my own understanding, are both nodepool already created at this point? or we're passing the nodepool names and `gkeInterp.createCluster` will create them concurrently if they don't exist?

Reply:
the nodepools don't exist yet -- we are passing them to GKE, and they will be created as part of cluster creation.